tests/Unit/IO/Test_H5.py

```diff
@@ -79,5 +78,16 @@ def test_get_header(self):
         self.assertEqual(datfile.get_header()[0:16], "#\n# File created")
         file_spec.close()
 
+    def test_groups(self):
+        file_spec = spectre_h5.H5File(self.file_name, 1)
+        file_spec.insert_dat("/element_data", ["Time", "Value"], 0)
+        file_spec.insert_dat("/element_position", ["x", "y", "z"], 0)
+        file_spec.insert_dat("/element_size", ["Time", "Size"], 0)
+        groups_spec = ["element_data.dat", "element_position.dat",
```

Comment:
I'm not sure we can rely on the order here. I think hdf5 doesn't make any guarantees about ordering of groups and has changed it between versions. What should work is something like:
```python
for group_name in group_spec:
    self.assert(group_name in file_spec.groups())
```